velox/exec/TraceUtil.cpp

```diff
@@ -164,6 +164,20 @@ RowTypePtr getDataType(
   return traceNode->sources().at(sourceIndex)->outputType();
 }
 
+std::vector<uint32_t> listPipelineIds(
+    const std::string& nodeTraceDir,
+    const std::shared_ptr<filesystems::FileSystem>& fs) {
+  const auto pipelineDirs = fs->list(nodeTraceDir);
+  std::vector<uint32_t> pipelineIds;
+  pipelineIds.reserve(pipelineDirs.size());
+  for (const auto& pipelineDir : pipelineDirs) {
+    pipelineIds.emplace_back(
```

Comment:
try and throw velox exception here with a better logging in error message? Thanks!